py/torch_tensorrt/dynamo/runtime/_WrapperTorchTensorRTModule.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import annotations
+
+import logging
+from contextlib import nullcontext
+from tempfile import tempdir
+from typing import List, Optional, Sequence, Tuple
+
+import torch
+import torch_tensorrt
+from torch._subclasses.fake_tensor import FakeTensorMode
+from torch.fx.experimental.proxy_tensor import unset_fake_temporarily
+from torch_tensorrt.dynamo import partitioning
+from torch_tensorrt.dynamo.conversion import DYNAMIC_DIM
+from torch_tensorrt.dynamo.utils import input_is_dynamic
+from torch_tensorrt.runtime._utils import _is_switch_required, _select_rt_device
+
+logger = logging.getLogger(__name__)
+
+
+class WrapperTorchTensorRTModule(torch.nn.Module):  # type: ignore[misc]
+    """This Wrapper runtime module is to record/replay whole cuda graph in sub modules
+
+    Args:
+        original_module: Unmodified FX GraphModule
+        compiled_module: Complied fx graphModule that will be wrapped
+        output_shapes: Shapes of output Tensors of the graph
+        output_dtypes: Output data types of the graph
+    Returns:
+        Output tensor or tensor list
+    """
+
+    def __init__(
+        self,
+        original_module: torch.nn.Module,
+        compiled_module: torch.nn.Module,
+        output_shapes: List[torch.Size],
+        output_dtypes: List[torch.dtype],
+    ):
+        super(WrapperTorchTensorRTModule, self).__init__()
+        self.original_module = original_module
+        self.compiled_module = compiled_module
+        self.inputs = partitioning.construct_submodule_inputs(compiled_module)
+        self.output_shapes = output_shapes
+        self.output_dtypes = output_dtypes
+
+        self._input_buffers: List[torch.Tensor] = []
+        self._output_buffers: List[torch.Tensor] = []
+        self.cudagraph: Optional[torch.cuda.CUDAGraph] = None
+        self.shape_key: Optional[str] = None
+        self.profiling_enabled = False
+        self.cudagraphs_enabled = False
+        self._caller_stream: Optional[torch.cuda.Stream] = None
+        self._engine_stream: Optional[torch.cuda.Stream] = None
+        self.input_is_dynamic = input_is_dynamic(self.inputs)
+
+        # Disable cudagrphs in submodules as it will be enabled in wrapper
+        for name, rt_mod in self.compiled_module.named_children():
+            if "_run_on_acc" in name:
+                rt_mod.set_whole_cudagraphs(True)
+
+        # Warm up is necessary to ensure that memory allocations and initializations are not recorded in cuda graphs
+        with unset_fake_temporarily():
+            inputs_tensor = [spec.torch_tensor.cuda() for spec in self.inputs]
+            s = torch.cuda.Stream()
+            s.wait_stream(torch.cuda.current_stream())
+            with torch.cuda.stream(s):
+                for _ in range(3):
+                    self.compiled_module(*inputs_tensor)
```

Comment:
should we wrap only the warm up runs in a error log level ? The scenario I'm thinking is let's say user ran his model in debug mode using 
```py
with torch_tensorrt.logging.debug():
      trt_model = torch_tensorrt.compile(model, inputs, debug=True)
```
This would print the warm up logs as well which can be confusing.